src/main/kotlin/it/pagopa/ecommerce/transactions/functions/timertriggers/handlers/PendingTransactionTimerTriggerHandler.kt

```diff
@@ -0,0 +1,75 @@
+package it.pagopa.ecommerce.transactions.functions.timertriggers.handlers
+
+import it.pagopa.ecommerce.transactions.functions.timertriggers.handlers.pojo.TimerTriggerExecutionContext
+import it.pagopa.ecommerce.transactions.functions.transactionanalyzer.PendingTransactionAnalyzer
+import java.time.LocalDateTime
+import java.time.temporal.ChronoUnit
+import java.util.function.Function
+import java.util.logging.Level
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.scheduling.support.CronExpression
+import org.springframework.stereotype.Component
+
+@Component("pendingTransactionAnalyzerHandler")
+class PendingTransactionTimerTriggerHandler : Function<TimerTriggerExecutionContext, Boolean> {
+
+    @Autowired private lateinit var pendingTransactionAnalyzer: PendingTransactionAnalyzer
+
+    @Value("\${pendingTransactions.batch.scheduledChron}")
+    private lateinit var chronExpression: String
+
+    @Value("\${pendingTransactions.batch.transactionsAnalyzer.executionRateMultiplier}")
+    private var executionRateMultiplier: Int = 1
```

Comment:
Can we refactor this with constructor injection?

```suggestion
class PendingTransactionTimerTriggerHandler : Function<TimerTriggerExecutionContext, Boolean>(
    @Autowired private val pendingTransactionAnalyzer: PendingTransactionAnalyzer
    @Value("\${pendingTransactions.batch.scheduledChron}") private val chronExpression: String,
    @Value("\${pendingTransactions.batch.transactionsAnalyzer.executionRateMultiplier}") private val executionRateMultiplier: Int = 1,
) {
```

Reply:
Yes, right👍🏽 fixed into https://github.com/pagopa/pagopa-ecommerce-transactions-functions/pull/9/commits/34fcf8f6219ab1c7819f33dd2ec1db97c48f588a